src/modules/exchangeProvider.js

```diff
@@ -92,7 +96,11 @@ const onCancelledOrder = function(orderId, maker, tokenAmount, weiAmount) {
     store.dispatch(refreshExchange());
     store.dispatch(refreshOrders());
     const userAccount = store.getState().web3Connect.userAccount;
-    store.dispatch(processNewTrade('CancelledOrder', userAccount, this));
+    if (userAccount === maker) {
+        store.dispatch(processNewTrade('CancelledOrder', userAccount, this));
+    } else {
+        processNewTrade('CancelledOrder', userAccount, this)
```

Comment:
I'm not sure what are you trying to achieve here? Why are you calling processNewTrade directly without dispatching? I think this should be enough:
```js
if (userAccount === maker) {
        store.dispatch(processNewTrade('CancelledOrder', userAccount, this));
}
```
NB: I'm not sure about account capitalisation, you might need .toLowerCase() on userAccount or maker if one of them is in all lowercase ([read more about account checksum](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md))

Reply:
okay, I understand and I made a fix.